marimo/_save/loaders/loader.py

```diff
@@ -113,10 +126,10 @@ def cache_attempt(
             )
         loaded = self.load_cache(hashed_context, cache_type)
         # TODO: Consider more robust verification
-        assert loaded.hash == hashed_context, INCONSISTENT_CACHE_BOILER_PLATE
-        assert set(defs | stateful_refs) == set(loaded.defs), (
-            INCONSISTENT_CACHE_BOILER_PLATE
-        )
+        if loaded.hash != hashed_context:
+            raise LoaderError("Hash mismatch in loaded cache.")
+        if set(defs | stateful_refs) != set(loaded.defs):
```

Comment:
Might be misunderstanding something, but the `set` is not needed right?

```suggestion
        if (defs | stateful_refs) != set(loaded.defs):
```